src/amr/data/field/field_geometry.hpp

```diff
@@ -184,74 +190,19 @@ namespace amr
             // box.upper is still box.lower + end-start, end &start of ghosts
 
 
-            if (!withGhost)
-            {
-                std::int32_t xStart = layout.physicalStartIndex(qty, core::Direction::X);
-                std::int32_t xEnd   = layout.physicalEndIndex(qty, core::Direction::X);
-
-                box.setLower(dirX, lower[dirX]);
-                box.setUpper(dirX, xEnd - xStart + lower[dirX]);
-
-                if (dimension > 1)
-                {
-                    std::int32_t yStart = layout.physicalStartIndex(qty, core::Direction::Y);
-                    std::int32_t yEnd   = layout.physicalEndIndex(qty, core::Direction::Y);
-
-                    box.setLower(dirY, lower[dirY]);
-                    box.setUpper(dirY, yEnd - yStart + lower[dirY]);
-                }
-                if (dimension > 2)
-                {
-                    std::int32_t zStart = layout.physicalStartIndex(qty, core::Direction::Z);
-                    std::int32_t zEnd   = layout.physicalEndIndex(qty, core::Direction::Z);
-
-                    box.setLower(dirZ, lower[dirZ]);
-                    box.setUpper(dirZ, zEnd - zStart + lower[dirZ]);
-                }
-            } // end withoutGhosts
 
+            box.setLower(dirX, lower[dirX]);
```

Comment:
this could be handled with a 

`
for_N<dim>([&](auto i){
    box.setLower(i, lower[i]);
    box.setUpper(i, upper[i] + layout.isPrimal(qty, static_cast<core::Direction>(i)));
})
`

Reply:
you can also get the centerings as an array and not need this new `isPrimal` function I think

Reply:
if you update it like `enum class Direction : std::uint8_t { X = 0, Y, Z };` you should be able to

Reply:
even as is it should work https://coliru.stacked-crooked.com/a/18587de1a1512534